Fabric/Views/NodeRegisitryView.swift

```diff
@@ -120,4 +134,41 @@ public struct NodeRegisitryView: View {
         let nodesForType:[NodeClassWrapper] = availableNodes.filter( { $0.nodeType == nodeType })
         return  self.searchString.isEmpty ? nodesForType : nodesForType.filter {  $0.nodeName.localizedCaseInsensitiveContains(self.searchString) }
     }
+    
+    /// Checks if a node's bounds are visible in the current viewport
+    /// Returns true only if the entire node is within the visible area
+    private func isNodeVisible(center: CGPoint, size: CGSize) -> Bool
```

Comment:
This func feels like it belongs on `Node` model, not in nodeRegistry, as it might be helpful elsewhere, or in a layout manager class which encapsulates some of the layout calcs used 'across' views.